src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java

```diff
@@ -23,88 +23,106 @@
-        userService.update(email, email, newPassword, user.getDiscogsUserName());
-        passwordRecoveryDao.deleteById(recoveryToken.getId());
+        try {
+            userService.update(email, email, newPassword, user.getDiscogsUserName());
+            passwordRecoveryDao.deleteById(recoveryToken.getId());
+        } catch (DataAccessException e) {
+            throw new RuntimeException(PasswordRecoveryError.UPDATE_PASSWORD_ERROR.getMessage());
+        }
     }
 
     @Override
     public void checkToken(String token) {
-        UUID tokenUUID = stringToUUD(token);
+        UUID tokenUUID = stringToUUID(token);
         RecoveryToken recoveryToken = passwordRecoveryDao.findByToken(tokenUUID)
-                .orElseThrow(() -> new RecoveryPasswordException(ErrorRecoveryPassword.TOKEN_NOT_FOUND_IN_DB.getMessage()));
+                .orElseThrow(() -> new RuntimeException(PasswordRecoveryError.TOKEN_NOT_FOUND_IN_DB.getMessage()));
         LocalDateTime tokenLifetime = recoveryToken.getCreatedAt().toLocalDateTime().plusHours(liveTokenHours);
         if (tokenLifetime.compareTo(LocalDateTime.now()) < 0) {
             log.debug("Token lifetime has come to an end.");
-            passwordRecoveryDao.deleteById(recoveryToken.getId());
-            throw new RecoveryPasswordException(ErrorRecoveryPassword.TOKEN_IS_EXPIRED.getMessage());
+            try {
+                passwordRecoveryDao.deleteById(recoveryToken.getId());
+            } catch (DataAccessException e) {
+                throw new RuntimeException(PasswordRecoveryError.DELETE_TOKEN_ERROR.getMessage());
+            }
+            throw new RuntimeException(PasswordRecoveryError.TOKEN_IS_EXPIRED.getMessage());
         }
     }
 
     @Override
     @Transactional
     public void sendLink(String email) {
-        checkForNotNull(email, ErrorRecoveryPassword.EMPTY_EMAIL);
+        checkForIsNotEmptyNotNull(email, PasswordRecoveryError.EMPTY_EMAIL);
         User user = userService.findByEmail(email)
-                .orElseThrow(() -> new RecoveryPasswordException(ErrorRecoveryPassword.EMAIL_NOT_FOUND_IN_DB.getMessage()));
-        RecoveryToken recoveryToken = addRecoveryUserToken(user.getId());
+                .orElseThrow(() -> new RuntimeException(PasswordRecoveryError.EMAIL_NOT_FOUND_IN_DB.getMessage()));
+        RecoveryToken recoveryToken = addRecoveryTokenWithUserId(user.getId());
         sendEmailWithLink(email, recoveryToken.getToken().toString());
     }
 
-    UUID stringToUUD(String token) {
+    UUID stringToUUID(String token) {
         try {
             return UUID.fromString(token);
         } catch (IllegalArgumentException e) {
-            throw new RecoveryPasswordException(ErrorRecoveryPassword.TOKEN_NOT_CORRECT_UUID.getMessage());
+            throw new RuntimeException(PasswordRecoveryError.TOKEN_NOT_CORRECT_UUID.getMessage());
         }
     }
 
     void sendEmailWithLink(String email, String recoveryToken) {
```

Comment:
Should have "private" access modfiier. Never consider things like "it could be helpful when...". For the moment it is being used only inside this class

Reply:
it is used in tests

Reply:
Why dont we test general method? We can test utility methods threw it 

For instance:

changePassword is using stringToUUID

You could create test case where during execution fo changePassword other method like stringToUUID should throw exception